pyproject.toml

```diff
@@ -59,10 +59,26 @@ exclude_lines = [
 ]
 
 [tool.pyright]
-include = ["examples", "hikari"]
+include = ["hikari"]
+exclude = ["**/__init__.py", "hikari/internal/enums.py", "hikari/internal/fast_protocol.py"]
 pythonVersion = "3.8"
 typeCheckingMode = "strict"
 
+reportMissingTypeStubs = "none"
+reportImportCycles = "none" # Doesn't account for TYPE_CHECKING
+reportIncompatibleMethodOverride = "none"  # This relies on ordering for keyword-only arguments
+reportOverlappingOverload = "none" # Type-Vars in last overloads may interfere
+reportIncompatibleVariableOverride = "none" # Cannot overwrite abstract properties using attrs
+
+# Attrs validators will always be unknown
+reportUnknownMemberType = "warning"
+reportUntypedFunctionDecorator = "warning"
+reportOptionalMemberAccess = "warning"
+
+# for mypy
+reportUnnecessaryIsInstance = "none"
```

Comment:
For all these, we might as well fix them instead of ignoring them, if not, they will just keep flying under the radar and we will have no idea of them

Reply:
The whole point of this Pull Request is to remove those that are completely unfixable. You cannot fix them unless you completely drop attrs which won't happen.

Reply:
There is an issue open to improve attrs support for pyright: https://github.com/python-attrs/attrs/issues/795, so this will most probably change in the future

Reply:
Great, we can re-enable them once support is rolled out.